From e0e032794a55aeaa39db8e566c9831dbd933b656 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:06 -0400
Subject: [ethernet] i40e: don't use OR to check a value

Message-id: <1409658042-22689-126-git-send-email-sassmann@redhat.com>
Patchwork-id: 90325
O-Subject: [RHEL7.1 PATCH 125/281] i40e: don't use OR to check a value
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jesse Brandeburg <jesse.brandeburg@intel.com>

A couple of FD checks ended up using bitwise OR to check
a value, which ends up always being evaluated to true.

This should fix the issue.  Thanks to DaveJ for noticing
and reporting the issue!

CC: Dave Jones <davej@redhat.com>
Reported-by: Dave Jones <davej@redhat.com>
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit d3a90b70d8ffe6dd82a3fa22515a2ebc399aca28)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_txrx.c b/drivers/net/ethernet/intel/i40e/i40e_txrx.c
index 5a22187..1fe28ca 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_txrx.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_txrx.c
@@ -458,13 +458,13 @@ static void i40e_fd_handle_status(struct i40e_ring *rx_ring,
    */
   if (fcnt_prog >= (fcnt_avail - I40E_FDIR_BUFFER_FULL_MARGIN)) {
    /* Turn off ATR first */
-   if (pf->flags | I40E_FLAG_FD_ATR_ENABLED) {
+   if (pf->flags & I40E_FLAG_FD_ATR_ENABLED) {
     pf->flags &= ~I40E_FLAG_FD_ATR_ENABLED;
     dev_warn(&pdev->dev, "FD filter space full, ATR for further flows will be turned off\n");
     pf->auto_disable_flags |=
              I40E_FLAG_FD_ATR_ENABLED;
     pf->flags |= I40E_FLAG_FDIR_REQUIRES_REINIT;
-   } else if (pf->flags | I40E_FLAG_FD_SB_ENABLED) {
+   } else if (pf->flags & I40E_FLAG_FD_SB_ENABLED) {
     pf->flags &= ~I40E_FLAG_FD_SB_ENABLED;
     dev_warn(&pdev->dev, "FD filter space full, new ntuple rules will not be added\n");
     pf->auto_disable_flags |=
-- 
1.7.1